### DISCUSSION TITLE
Feature request: comment on any line in a Pull Request

### DISCUSSION THREAD

**Discussion by michcio1234** [community/community#23507] [Pull Requests]
<h2>
<a name="what-1" class="anchor" href="#what-1"></a>What</h2>
<p>I would like to be able to add a line comment to <strong>any</strong> code line in a Pull Request, not only to the changed lines and their surrounding.</p>
<h2>
<a name="why-2" class="anchor" href="#why-2"></a>Why</h2>
<p>When reviewing a PR, I often have thoughts on lines that were not changed.</p>
<p>E.g. I want to suggest changing a variable name, and I’d like to do it by adding a suggestion on each line where this variable is used, so that I (or a PR author) can commit the changes directly from Github, without going back to their IDE. If a variable is used in an unchanged line, I can’t do it.</p>
<p>Another example is when you review a change in a long forgotten file and you realise that actually more changes are needed. They are often in unchanged lines, which currently cannot be commented on.</p>
<h2>
<a name="how-3" class="anchor" href="#how-3"></a>How</h2>
<p>This could be done the exact same way as line- or range-commenting currently works.</p>

**Comment by Akash1134** [comment, marked as answer]
<p>Hello - and thank you for this feedback!</p>
<p>Please submit your feedback via the new GitHub <a href="https://github.com/github/feedback/discussions/categories/pull-requests-feedback">Feedback Discussions forum - Pull Request</a> so that our product team can better track your request :slight_smile:. It’s a new thing, but we’re trying to push all product feedback to the same new area so it’s easier to find and consolidate user feedback!</p>

**Comment by michcio1234** [comment]
<p>Oh, thank you, I wasn’t aware of that place. Maybe it can be exposed better on <a href="https://support.github.com/" rel="noopener nofollow ugc">https://support.github.com/</a>? Anyway, this thread can be closed.</p>

**Comment by queenofcorgis** [comment]
<p>Hi <a class="mention" href="/michcio1234">@michcio1234</a> - the link <a class="mention" href="/akash1134">@Akash1134</a> shared with you is the best place for it as our Support site is for account-specific issues, data queries, etc. Let us know if you have any questions!</p>

**Comment by mfrancis44** [comment]
@michcio1234 have you checked out Asronuts Github app? They're using AI to do line by line comments, especially for errors spotted in your pull request. Here you go https://github.com/marketplace/astronuts-app
